src/credential_provider.rs

```diff
@@ -0,0 +1,432 @@
+use crate::{MomentoError, MomentoResult};
+use base64::Engine;
+use jsonwebtoken::{decode, Algorithm, DecodingKey, Validation};
+use serde::{Deserialize, Serialize};
+use std::env;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct JwtClaims {
+    sub: String,
+    c: Option<String>,
+    cp: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct V1Token {
+    pub api_key: String,
+    pub endpoint: String,
+}
+
+const CACHE_ENDPOINT_PREFIX: &str = "cache.";
+const CONTROL_ENDPOINT_PREFIX: &str = "control.";
+
+#[derive(Debug, Clone)]
+pub struct CredentialProvider {
+    pub auth_token: String,
```

Comment:
Good call. I don't think `V1Token` needs `Debug` on it - it's not public and nothing is currently using it. `CredentialProvider` does need it because of how tests are written, and although  `CredentialProviderBuilder` doesn't need it for tests it seems probably nice to provide it since it's public.

Reply:
Fixed in [ed9fe51](https://github.com/momentohq/client-sdk-rust/pull/146/commits/ed9fe519693969269ef07187f90f851b68a9e689)